npm/design-system/src/components/FileExplorer/FileExplorer.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react'
+import { makeFileHierarchy } from './helpers/makeFileHierarchy'
+import { FileExplorerProps, FileTreeProps, FileLike, FolderOrFile } from './types'
+import styles from './FileExplorer.module.scss'
+import { InlineIcon } from '@iconify/react'
+import javascriptIcon from '@iconify/icons-vscode-icons/file-type-js-official'
+import typescriptIcon from '@iconify/icons-vscode-icons/file-type-typescript-official'
+import reactJs from '@iconify/icons-vscode-icons/file-type-reactjs'
+import reactTs from '@iconify/icons-vscode-icons/file-type-reactts'
+import folderClosed from '@iconify/icons-vscode-icons/default-folder'
+import folderOpen from '@iconify/icons-vscode-icons/default-folder-opened'
+import { library } from '@fortawesome/fontawesome-svg-core'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { fas } from '@fortawesome/free-solid-svg-icons'
+import cs from 'classnames'
+
+library.add(fas)
+
+const icons: Record<string, any> = {
+  js: { icon: javascriptIcon },
+  ts: { icon: typescriptIcon },
+  tsx: { icon: reactTs },
+  jsx: { icon: reactJs },
+  folderOpen: { icon: folderOpen },
+  folderClosed: { icon: folderClosed },
+}
+
+const getExt = (path: string) => {
+  const extensionMatches = path.match(/(?:\.([^.]+))?$/)
+
+  return extensionMatches ? extensionMatches[1] : ''
+}
+
+export const FileExplorer: React.FC<FileExplorerProps> = (props) => {
+  return (<nav className={cs(props.className || '', styles.nav)}><FileTree files={makeFileHierarchy(props.files) || []} isSelected={props.isSelected || (() => {})} onClick={props.onClick || (() => {})}></FileTree></nav>)
+}
+
+export const FileTree: React.FC<FileTreeProps> = (props) => {
+  const depth = props.depth || 0
+  const [files, setFiles] = React.useState(props.files || [])
```

Comment:
This seems incorrect - now we have two sources of truth. Why do we need two sources of truth? Can't we just use `props.files`? Design system should not have internal state, ideally everything should be a function of props

Reply:
Should I be passing a callback all the way down and making the top-level component mutate files?

Reply:
Yes

Reply:
> > > Also, it is weird that components like FileTree are inside the design system. Cause it looks like more "end" component, maybe better place will be inside runner-ct?
> > 
> > 
> > I think it makes sense because we will use it in runner-ct as well as with the node UI (possibly desktop-gui?). There's also some other benefits.
> 
> Technically we should have separate design system and Cypress component library folders (where the only components the design system has are individual atoms), but I don't know that it's worth going to that extreme now.

Yes. You're correct. I'd be happy just to get our design tokens plumbed first and also keep reusable UI patterns separate from domain models.